compiler/qsc_passes/src/entry_point/tests.rs

```diff
@@ -51,13 +51,43 @@ fn test_entry_point_attr_to_expr() {
 }
 
 #[test]
-fn test_entry_point_attr_missing() {
+fn test_entry_point_attr_missing_implies_main() {
     check(
         indoc! {"
             namespace Test {
                 operation Main() : Int { 41 + 1 }
             }"},
         "",
+        &expect![[r#"
+            Expr 12 [0-0] [Type Int]: Call:
+                Expr 11 [32-36] [Type Int]: Var: Item 1
+                Expr 10 [36-38] [Type Unit]: Unit"#]],
+    );
+}
+
+#[test]
+fn test_entry_point_attr_missing_implies_main_alernate_casing() {
```

Comment:
Looking here, maybe it is allowing any casing. Seems like an odd choice, being we are a case sensitive language and 'opinionated' in much of our coding guidelines. Any reason we don't require `Main` casing?

Reply:
I could go either way on this. On the one hand, the fact that items are pascal cased and variables are camel cased is in the style guide but not enforced at compiling. This would be the first place where we are requiring a user follow that guidance. On the other hand, it's more consistent to require it rather than treat all casings of `Main` as equal. It's an easy thing to remove, we just need to decide on a preference for behavior.

Reply:
Looking around, it seems both Rust and C# are strongly opinionated on this point and accept only `main` and `Main` respectively, with any other casing being an error. That seems like good enough reason to me to make ours case sensistive too. I'll update.